tests/ui/consts/required-consts/collect-in-dead-fn.rs

```diff
@@ -1,15 +1,11 @@
 //@revisions: noopt opt
-//@[noopt] build-fail
+//@ build-fail
 //@[opt] compile-flags: -O
```

Comment:
The way that compiletest calculates the opt level for tests depends on several factors (`optimize-tests` setting, whether it's a build/fail/run test).
I think it's better to be explicitly and set `-Copt-level=0` for noopt tests to override heuristics.

Reply:
Uff, okay I thought the opt level defaults to 0. I don't think the test suite will pass with other settings -- they are clearly not covered by CI so they will ~never work.

So I think we should remove such settings and heuristics, they are fragile and cause undue maintenance costs.

Reply:
In merge-CI the testsuite is run with `optimize-tests` true and false at least but for UI tests that only affects run tests, not build tests I think.

Anyway, if the test is trying to cover specific optimization levels it should be part of the test configuration. Things accidentally depending on defaults of options has caused problems in the past when the defaults changed. E.g. tests testing CGU-dependent things failing with 1CGU.

The majority of UI tests aren't trying to test things that depend on specific compiler options.